src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1142,6 +1142,7 @@ public Collection<Object> createComponents(
         final XFFResolver xffResolver = new XFFResolver(threadPool);
         backendRegistry = new BackendRegistry(settings, adminDns, xffResolver, auditLog, threadPool, cih);
         backendRegistry.registerClusterSettingsChangeListener(clusterService.getClusterSettings());
+        cr.subscribeOnChange(configMap -> { backendRegistry.invalidateCache(); });
```

Comment:
why add a subscriber here instead of invalidating cache on config reload?

Reply:
As we combine concurrent reload actions into one, doing the cache flush on the action level would flush the cache potentially redundantly several times (for each queued reload).